src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -495,6 +502,10 @@ private function createFirewall(ContainerBuilder $container, string $id, array $
                 ->addTag('kernel.event_subscriber', ['dispatcher' => $firewallEventDispatcherId]);
 
             $listeners[] = new Reference('security.firewall.authenticator.'.$id);
+
+            // Add authenticators to the debug:firewall command
+            $debugCommand = $container->getDefinition('security.command.debug_firewall');
+            $debugCommand->replaceArgument(3, array_merge($debugCommand->getArgument(3), [$id => $authenticators]));
```

Comment:
```suggestion
            $authenticatorsLocator = $container->getDefinition('security.authenticator.authenticators_locator');
            $authenticatorsLocator->replaceArgument(0, array_merge($authenticatorsLocator->getArgument(0), [$id => array_map(function ($ref) {
                return new new ServiceClosureArgument($ref);
            }, $authenticators)]));
```

Reply:
I can't get this to work, since the locator shouldn't return a single service (authenticator) per firewall, but an array of them (a single firewall can have multiple authenticators).

I run into a "Function name must be a string" in ServiceLocatorTrait on line 70 (`return $this->factories[$id]($this)`).

The same happens when I try to fill a service locator with service iterators (those end up as a `Symfony\Component\DependencyInjection\Argument\RewindableGenerator`, which is not a callable).